File: solutions_2652486_0/C++/logan12358/luck.cpp
#include <cstdio>

int main() {
scanf("%*d");
printf("Case #1:\n");
    int r, n, m, k; scanf("%d %d %d %d", &r, &n, &m, &k);
    for(int i=0; i<r; i++) {
        int product[500];
        for(int j=0; j<500; j++) product[j]=0;
        for(int j=0; j<k; j++) {
            int x; scanf("%d", &x);
            product[x]++;
        }
        int best=0;
        int A, B, C;
        for(int a=2; a<=m; a++) {
            for(int b=2; b<=m; b++) {
                for(int c=2; c<=m; c++) {
                    int score=0;
                    bool f = false;
                    for(int j=0; j<500; j++) {
                        if(product[j]>0) {
                            if(j==1 or j==a or j==b or j==c or j==a*b or j==a*c or j==c*b or j==a*b*c);
                            else f=true;
                        }
                    }
                    if(f) continue;
                    for(int aa=0; aa<2; aa++) {
                        for(int bb=0; bb<2; bb++) {
                            for(int cc=0; cc<2; cc++) {
                                score += product[(aa?a:1)*(bb?b:1)*(cc?c:1)];
                            }
                        }
                    }
                    if(score>best) {
                        best=score;
                        A=a;
                        B=b;
                        C=c;
                    }
                }
            }
        }
        printf("%d%d%d\n", A, B, C);
    }
}
                    

File: solutions_2652486_0/Python/sharpobjectstopstealingmyname/luck.py
T = int(raw_input())

unit = 1/8.0
p = [[[[0]*200 for x in xrange(6)]for y in xrange(6)] for z in xrange(6)]
for x in xrange(2,6):
    for y in xrange(2,6):
        for z in xrange(2,6):
            p[x][y][z][1] += unit
            p[x][y][z][x] += unit
            p[x][y][z][y] += unit
            p[x][y][z][z] += unit
            p[x][y][z][x*y] += unit
            p[x][y][z][x*z] += unit
            p[x][y][z][y*z] += unit
            p[x][y][z][x*y*z] += unit

for qq in xrange(1,T+1):
    r,n,m,k = map(int,raw_input().split(" "))
    print "Case #1:"
    for case in xrange(r):
        stuff = map(int, raw_input().split(" "))
        best = "222"
        score = 0
        for x in xrange(2,6):
            for y in xrange(2,6):
                for z in xrange(2,6):
                    can_score = reduce(lambda a,b:a*b, map(lambda c: p[x][y][z][c], stuff))
                    if can_score > score:
                        score = can_score
                        best = str(x)+str(y)+str(z)
        print(best)
